domain/src/main/kotlin/no/nav/su/se/bakover/domain/kontrollsamtale/Kontrollsamtale.kt

```diff
@@ -0,0 +1,63 @@
+package no.nav.su.se.bakover.domain.kontrollsamtale
+
+import no.nav.su.se.bakover.common.Tidspunkt
+import no.nav.su.se.bakover.common.endOfMonth
+import no.nav.su.se.bakover.common.periode.Periode
+import no.nav.su.se.bakover.common.startOfMonth
+import no.nav.su.se.bakover.common.zoneIdOslo
+import no.nav.su.se.bakover.domain.vedtak.Vedtak
+import java.time.Clock
+import java.time.LocalDate
+import java.util.UUID
+
+data class Kontrollsamtale(
+    val id: UUID = UUID.randomUUID(),
+    val opprettet: Tidspunkt = Tidspunkt.now(),
+    val sakId: UUID,
+    val innkallingsdato: LocalDate?,
+    val status: Kontrollsamtalestatus,
+    val frist: LocalDate?
+) {
+    companion object {
+        fun fromVedtak(vedtak: Vedtak): Kontrollsamtale {
+            val innkallingsdato = regnUtInnkallingsdato(vedtak.periode, vedtak.opprettet.toLocalDate(zoneIdOslo))
+
+            return Kontrollsamtale(
+                sakId = vedtak.behandling.sakId,
+                innkallingsdato = innkallingsdato,
+                status = Kontrollsamtalestatus.PLANLAGT_INNKALLING,
+                frist = innkallingsdato?.endOfMonth()
+            )
+        }
+    }
+}
+
+enum class Kontrollsamtalestatus(val value: String) {
+    PLANLAGT_INNKALLING("PLANLAGT_INNKALLING"),
+    INNKALT("INNKALT"),
+    GJENNOMFØRT("GJENNOMFØRT")
+}
+
+private fun LocalDate.isLessThanOneMonthAfter(localDate: LocalDate) = this.isBefore(localDate.plusMonths(1))
+
+fun regnUtInnkallingsdato(periode: Periode, vedtaksdato: LocalDate, clock: Clock? = null): LocalDate? {
```

Comment:
Antar denne er public pgr. testene? Kan vi endre så clock ikke er frivllig :pray:

Reply:
Er grunnen til at vi tar inn clock i andre funksjoner pga at vi skal kunne skrive tester? Har antatt det hvert fall 😅 Derfor jeg også lot den være nullable. Virker litt tilfeldig hva som tar inn clock og ikke (gjetter at det er nettopp pga testdekning), så ser ikke helt grunnen til at vi skal sende inn en clock i prod-kode. Kan godt gjøre det dog, men er det en grunn til at den skal sendes inn?😊

Reply:
Det er for tester ja. Vi har og noen regresjonstester/black-box tester som går helt fra web-laget. Der propageres klokka fra Application.kt og innover. Så ideelt sett for testene sin del bør vi ikke bruke now() og kun en plass i Application.kt bruke Clock.systemUTC() og la den propagere. Når det er sagt sliter tidvis service-testene med det samme.